packages/flutter_tools/lib/src/version.dart

```diff
@@ -729,18 +747,73 @@ class GitTagVersion {
         _runGit('git fetch $_flutterGit --tags', processUtils, workingDirectory);
       }
     }
-    return parse(_runGit('git describe --match v*.*.* --first-parent --long --tags', processUtils, workingDirectory));
+    // `--match` glob must match old version tag `v1.2.3` and new `1.2.3-dev.4.5`
+    return parse(_runGit('git describe --match *.*.* --first-parent --long --tags', processUtils, workingDirectory));
   }
 
-  static GitTagVersion parse(String version) {
-    final RegExp versionPattern = RegExp(r'^v([0-9]+)\.([0-9]+)\.([0-9]+)(?:\+hotfix\.([0-9]+))?-([0-9]+)-g([a-f0-9]+)$');
+  /// Check for the release tag format pre-v1.17.0
+  static GitTagVersion parseFirstVersion(String version) {
```

Comment:
Is there a point in time where it would be safe to remove the old tag parsing logic? If so, lets file an issue

Reply:
Good idea, filed https://github.com/flutter/flutter/issues/53850 to track.